File: using_js/smallest.js
function smallest_fn(arr){
    let sorted = [];
    let min = 0;
    for(let i=1; i<=arr.length-1; i++){
        if (arr[min]>arr[i]){
            min = i
        }
    }
    return arr[min]
}
console.log("min:"+smallest_fn([-1, -2, 1, 4, 8, 7, 3, 1, 2]))

File: using_rs/smallest.rs
fn smallest_fn(arr: &[i64]) -> i64{
    let mut min = arr[0];
    for &i in arr.iter() {
        if i<min{
            min = i;
        }
    }
    min
}

fn main(){
    let arr = [-1, -2, 1, 4, 8, 7, 3, 1, 2];
    println!("{} is min", smallest_fn(&arr));
}